lectures/three_dimensional_image_processing.ipynb

```diff
@@ -0,0 +1,864 @@
+    "closing = np.empty_like(rescaled)\n",
+    "dilation = np.empty_like(rescaled)\n",
+    "erosion = np.empty_like(rescaled)\n",
+    "opening = np.empty_like(rescaled)\n",
+    "\n",
+    "binary_closing = np.empty_like(binary)\n",
+    "binary_dilation = np.empty_like(binary)\n",
+    "binary_erosion = np.empty_like(binary)\n",
+    "binary_opening = np.empty_like(binary)\n",
+    "\n",
+    "selem = skimage.morphology.disk(3)\n",
+    "\n",
+    "for plane, (image, binary_image) in enumerate(zip(rescaled, binary)):\n",
+    "    closing[plane] = skimage.morphology.closing(image, selem)\n",
+    "    dilation[plane] = skimage.morphology.dilation(image, selem)\n",
+    "    erosion[plane] = skimage.morphology.erosion(image, selem)\n",
+    "    opening[plane] = skimage.morphology.opening(image, selem)\n",
+    "    \n",
+    "    binary_closing[plane] = skimage.morphology.binary_closing(binary_image, selem)\n",
+    "    binary_dilation[plane] = skimage.morphology.binary_dilation(binary_image, selem)\n",
+    "    binary_erosion[plane] = skimage.morphology.binary_erosion(binary_image, selem)\n",
+    "    binary_opening[plane] = skimage.morphology.binary_opening(binary_image, selem)\n",
+    "    \n",
+    "_, ((a, b, c, d), (e, f, g, h)) = plt.subplots(nrows=2, ncols=4, figsize=(16, 8))\n",
+    "\n",
+    "show_plane(a, closing[32], title=\"Closing\")\n",
+    "show_plane(b, opening[32], title=\"Opening\")\n",
+    "show_plane(c, erosion[32], title=\"Erosion\")\n",
+    "show_plane(d, dilation[32], title=\"Dilation\")\n",
+    "\n",
+    "show_plane(e, binary_closing[32], title=\"Binary closing\")\n",
+    "show_plane(f, binary_opening[32], title=\"Binary opening\")\n",
+    "show_plane(g, binary_erosion[32], title=\"Binary erosion\")\n",
+    "show_plane(h, binary_dilation[32], title=\"Binary dilation\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = 20\n",
+    "\n",
+    "remove_holes = skimage.morphology.remove_small_holes(\n",
+    "    binary, \n",
+    "    min_size=np.product(np.divide(a, spacing))\n",
+    ")\n",
+    "\n",
+    "display(remove_holes)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = 40\n",
+    "\n",
+    "remove_objects = skimage.morphology.remove_small_objects(\n",
+    "    remove_holes, \n",
+    "    min_size=np.product(np.divide(a, spacing))\n",
+    ")\n",
+    "\n",
+    "display(remove_objects)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Segmentation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def get_cmap(labels, name=\"viridis\"):\n",
+    "    cmap = matplotlib.cm.get_cmap(\"viridis\")\n",
+    "    masked_labels = np.ma.masked_where(labels == 0, labels)\n",
```

Comment:
This is the trick I know for setting the value 0 to black, regardless of colormap.